generic-ec/src/scalar.rs

```diff
@@ -142,25 +142,14 @@ impl<E: Curve> Scalar<E> {
 
     /// Interprets provided bytes as integer $i$ in big-endian order, returns scalar $s = i \mod q$
     pub fn from_be_bytes_mod_order(bytes: impl AsRef<[u8]>) -> Self {
-        let scalar_0x100 = Scalar::from(0x100_u16);
-        bytes
-            .as_ref()
-            .iter()
-            .fold(Scalar::<E>::zero(), |acc, byte| {
-                acc * scalar_0x100 + Scalar::from(*byte)
-            })
+        let scalar = E::Scalar::from_be_bytes_mod_order(bytes.as_ref());
+        Self::from_raw(scalar)
```

Comment:
I'm now surprised we ever came up with something so inefficient

Reply:
Though I'm aslo surprised that the bit-by-bit algorithm in multiscalar is efficient at all

Reply:
It was generic enough, there's no uniform support of reducing 256/512-bits integers mod order, which requires to write more code, and it wasn't that obvious that the code is very inefficient until I did benchmarks